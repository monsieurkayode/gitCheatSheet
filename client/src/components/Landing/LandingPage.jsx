import React, { Fragment } from 'react';
import styled from 'styled-components';
import Container from '../Common/Container';
import SearchBar from '../Common/SearchBar';
import CategoryCard from '../Common/CategoryCard';
import Footer from '../Common/Footer';
import categories from '../../../../global/categories';

const Section = styled.section`
  margin-top: 80px;
  display: flex;
  flex-flow: row wrap;

  @media only screen and (max-width: 424px) {
    margin-top: 40px;
  }
`;

const LandingPage = () => (
  <Fragment>
    <Container>
      <SearchBar />
      <Section>
        { categories.map(category => (
          <CategoryCard
            className="category"
            key={category}
            category={category}
          />
        ))}
      </Section>
    </Container>
    <Footer />
  </Fragment>
);

export default LandingPage;
