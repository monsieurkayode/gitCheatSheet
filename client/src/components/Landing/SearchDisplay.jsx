/* eslint-disable no-underscore-dangle */
import React, { Fragment } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CheatCard from '../Cheat/CheatCard';
import NoContent from '../Common/NoContent';

const Section = styled.section`
  column-count: auto;
  column-width: 360px;
  max-width: 1232px;
  margin: 60px 0;
  flex: 1;
  width: 100%;
`;

const NotFound = styled(NoContent)`
  flex: 1
`;

export const SearchDisplay = ({
  cheats,
  loading,
  searchTerm
}) => {
  const filteredCheats = cheats
    .filter((cheat) => {
      const keyword = searchTerm.toLowerCase();
      return (
        cheat.category.name.toLowerCase().includes(keyword)
        || cheat.keywords.includes(keyword)
        || cheat.header.toLowerCase().includes(keyword)
        || cheat.description.toLowerCase().includes(keyword)
      );
    });

  return (
    <Fragment>
      {!loading
        && filteredCheats.length === 0
        && <NotFound content="No cheat found, try with a diffrent keyword" />}
      {filteredCheats.length > 0 && searchTerm.trim().length > 0 && (
      <Section>
        { filteredCheats
          .map(cheat => <CheatCard key={cheat._id} {...cheat} />)}
      </Section>)}
    </Fragment>
  );
};

SearchDisplay.defaultProps = {
  searchTerm: ''
};

SearchDisplay.propTypes = {
  cheats: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  loading: PropTypes.bool.isRequired,
  searchTerm: PropTypes.string
};

const mapStateToProps = ({ cheatSheets }) => ({
  cheats: cheatSheets.cheats,
  loading: cheatSheets.makingApiRequest,
  searchTerm: cheatSheets.searchTerm.trim()
});

export default connect(mapStateToProps)(SearchDisplay);
