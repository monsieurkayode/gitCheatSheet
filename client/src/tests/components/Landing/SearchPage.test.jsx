import React from 'react';
import { shallow } from 'enzyme';
import { SearchDisplay } from '../../../components/Landing/SearchDisplay';

const setup = () => {
  const props = {
    cheats: [
      {
        category: { name: 'install' },
        keywords: [],
        description: 'install',
        _id: '1'
      }
    ],
    loading: false
  };

  return shallow(<SearchDisplay {...props} />);
};

describe('SearchDisplay Component', () => {
  const wrapper = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
