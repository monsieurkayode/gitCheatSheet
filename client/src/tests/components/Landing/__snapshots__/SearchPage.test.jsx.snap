// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDisplay Component should render correctly 1`] = `<Fragment />`;
